### PR TITLE
Endpoint Slice support for ServiceImport

### PR DESCRIPTION
**Pull request by tanujd11**
**What type of PR is this?**

Feature: Multicluster services API

**What this PR does / why we need it**:

With https://github.com/envoyproxy/gateway/pull/1494 EndpointSlice support is added for backends for gateway. This PR do the same for ServiceImport.

**Which issue(s) this PR fixes**:

Fixes https://github.com/envoyproxy/gateway/issues/1908

### CODE REVIEW

File: internal/gatewayapi/route.go [hunk 1]
```diff
@@ -1009,14 +1010,44 @@ func (t *Translator) processDestination(backendRef v1beta1.BackendRef,
...
 	switch KindDerefOr(backendRef.Kind, KindService) {
-	// TODO: Use EndpointSlice for ServiceImport
 	case KindServiceImport:
-		backendIps := resources.GetServiceImport(backendNamespace, string(backendRef.Name)).Spec.IPs
-		for _, ip := range backendIps {
-			ep := ir.NewDestEndpoint(
-				ip,
-				uint32(*backendRef.Port))
-			endpoints = append(endpoints, ep)
+		serviceImports := resources.GetServiceImport(backendNamespace, string(backendRef.Name))
```
Comment (arkodg) [inline internal/gatewayapi/route.go:1014]: shouldnt this be a single `serviceImport` ?

File: internal/gatewayapi/route.go [hunk 2]
```diff
@@ -1009,14 +1010,44 @@ func (t *Translator) processDestination(backendRef v1beta1.BackendRef,
...
+		for _, port := range serviceImports.Spec.Ports {
+			if port.Port == int32(*backendRef.Port) {
+				servicePort = port
+				break
+			}
+		}
+		if !t.EndpointRoutingDisabled {
+			endpointSlices := resources.GetEndpointSlicesForBackend(backendNamespace, string(backendRef.Name), KindDerefOr(backendRef.Kind, KindService))
+
+			for _, endpointSlice := range endpointSlices {
```
Comment (arkodg) [inline internal/gatewayapi/route.go:1025]: can we make this into a func so we dont need to duplicate the same code below as well ?

File: internal/gatewayapi/route.go (hunk 1 above)
Comment (tanujd11) [reply internal/gatewayapi/route.go:1014]: yeah it is a tpyo :/

File: internal/gatewayapi/route.go (hunk 2 above)
Comment (tanujd11) [reply internal/gatewayapi/route.go:1025]: sure

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zhaohuabing — APPROVED** [review]
LGTM. Great job!

**Pull request merged**